### PR TITLE
herokuapp homework tests

### PR DESCRIPTION
**Pull request by KaterinBaranova**
Please take a look at my herokuapp homework

### CODE REVIEW

File: src/test/java/herokuapp/AddDeleteElementsTests.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    }
+
+    @AfterClass
+    public void tearDown() {
+        driver.quit();
+    }
+
+    @Test
+    public void addElementTest() {
```
Comment (voropa) [inline src/test/java/herokuapp/AddDeleteElementsTests.java:32]: There should be a single test with following steps:
1. Add 2 elements
2. Delete one element
3. Verify elements count

Comment (voropa) [reply src/test/java/herokuapp/AddDeleteElementsTests.java:32]: These 2 are redundant: addElementTest, deleteElementTest.
So please delete them

File: src/test/java/herokuapp/DropdownTests.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @AfterClass
+    public void tearDown() {
+        driver.quit();
+    }
+
+    @Test
+    public void dropdownTest() {
+        Select select = new Select(driver.findElement(By.id("dropdown")));
```
Comment (voropa) [inline src/test/java/herokuapp/DropdownTests.java:34]: driver.get() action is missing

File: src/test/java/herokuapp/InputsTests.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        driver.quit();
+    }
+
+    @Test
+    public void inputsTest() {
+        driver.get("http://the-internet.herokuapp.com/inputs");
+        driver.findElement(By.xpath("//input")).sendKeys(Keys.ARROW_UP);
+        String upKeyResult = driver.findElement(By.xpath("//input")).getAttribute("value");
+        assertEquals(upKeyResult, "1");
+        driver.findElement(By.xpath("//input")).sendKeys(Keys.ARROW_DOWN, Keys.ARROW_DOWN, Keys.ARROW_DOWN);
```
Comment (voropa) [inline src/test/java/herokuapp/InputsTests.java:35]: You don't need to locate this element again and again until page is reloaded.
You can find it once at the very beginning and then reuse.

File: src/test/java/herokuapp/InputsTests.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    }
+
+    @AfterClass
+    public void tearDown() {
+        driver.quit();
+    }
+
+    @Test
+    public void inputsTest() {
```
Comment (voropa) [inline src/test/java/herokuapp/InputsTests.java:30]: Also you can split this test into 3 separate:
1. arrowsInputTest
2. stringInputTest
3. numericInputTest

File: src/test/java/herokuapp/DropdownTests.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        select.selectByValue("1");
```
Comment (voropa) [inline src/test/java/herokuapp/DropdownTests.java:35]: "Взять все элементы дроп-дауна и проверить их наличие." verification is missing.
Should be something  like:
Assert(options.count(), 2);
Assert(options.get(0).getText(), "Option 1")
Assert(options.get(0).getText(), "Option 2")

File: src/test/java/herokuapp/CheckboxesTests.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    @AfterClass
+    public void tearDown() {
+        driver.quit();
+    }
+
+    @Test
+    public void checkboxTest() {
+
+        driver.get("http://the-internet.herokuapp.com/checkboxes");
+        WebElement firstCheckbox = driver.findElement(By.cssSelector("[type=checkbox 1]"));
```
Comment (voropa) [inline src/test/java/herokuapp/CheckboxesTests.java:35]: This is incorrect locator, and it doesn't work. 

Please use driver.findElements() to  get a List of checkboxes and then access them by index e.g. checkboxes.get(0/1);

File: src/test/java/herokuapp/CheckboxesTests.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        Assert.assertNull(firstCheckbox.getAttribute("unchecked"));
```
Comment (voropa) [inline src/test/java/herokuapp/CheckboxesTests.java:36]: Checkbox doesn't have "unchecked" attribute. If it's unchecked then "checked" attribute will be null. 
So in order to verify the checkbox is unchecked you should use the following:
AssertNull(checkbox.getAttribute("checked"))

**Review by voropa — CHANGES_REQUESTED** [review]

**Comment by KaterinBaranova** [comment]
Fixed testes
not sure how to use correctly(dropdown)
Assert(options.count(), 2);
Assert(options.get(0).getText(), "Option 1")
Assert(options.get(0).getText(), "Option 2")

**Comment by voropa** [comment]
> Fixed testes not sure how to use correctly(dropdown) Assert(options.count(), 2); Assert(options.get(0).getText(), "Option 1") Assert(options.get(0).getText(), "Option 2")

Should be something like this:
Select select = new Select(driver.findElement(By.id("dropdown")));
        List<WebElement> selectOptions = select.getAllSelectedOptions();
        Assert.assertEquals(selectOptions.size(), 2);
        Assert.assertEquals(selectOptions.get(0).getText(), "Please select an option");
        Assert.assertEquals(selectOptions.get(1).getText(), "Option 1");
        Assert.assertEquals(selectOptions.get(2).getText(), "Option 2");
        
        // OR
        
        Select select = new Select(driver.findElement(By.id("dropdown")));
        List<WebElement> selectOptions = select.getAllSelectedOptions();
        List<String> expectedOptions = Arrays.asList("Please select an option", "Option 1", "Option 2");
        Assert.assertEquals(selectOptions.stream().map(WebElement::getText).toArray(), expectedOptions);

**Pull request closed without merging**